reward_hub/base.py

```diff
@@ -16,29 +16,48 @@
 from typing import Union, List
 from abc import ABC, abstractmethod
 import math
+from enum import Enum
 
 
+class AggregationMethod(Enum):
+    MODEL = "model"
+    PRODUCT = "prod"
+    MINIMUM = "min"
+    LAST = "last"
+
 class PRMResult:
     """
     full result of process reward model
     """
-    def __init__(self, scores: List[float], aggregate_method: str = None):
+    def __init__(self, scores: List[float], aggregation_method: Union[AggregationMethod, str] = AggregationMethod.LAST):
         self.step_scores = scores
         self.product = math.prod(scores)
         self.min = min(scores)
         self.last = scores[-1]
 
-        if aggregate_method == "prod":
+        # Handle both string and enum inputs for backward compatibility
+        if isinstance(aggregation_method, str):
+            try:
```

Comment:
this try block can be simplified as `aggregation_method  = AggregationMethod(aggregation_method)`